strawberryfields/api/devicespec.py

```diff
@@ -97,17 +99,35 @@ def gate_parameters(self):
 
         return gate_parameters
 
+    def layout_is_formatted(self):
+        """bool: Whether the device layout is formatted or not."""
+        p = re.compile(r"{{\w*}}")
+        return not bool(p.search(self.layout))
```

Comment:
Minor: perhaps this could be done with less overhead by simply returning `"{{" in self.layout`?